azurelinuxagent/common/utils/archive.py

```diff
@@ -48,10 +48,10 @@
 
 _GOAL_STATE_PATTERN = re.compile(r"^(.*)/GoalState\.(\d+)\.xml$", re.IGNORECASE)
 
-# 2018-04-06T08:21:37.142697
-# 2018-04-06T08:21:37.142697.zip
-_ARCHIVE_PATTERNS_DIRECTORY = re.compile(r"^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d+$")
-_ARCHIVE_PATTERNS_ZIP = re.compile(r"^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d+\.zip$")
+# 2018-04-06T08:21:37.142697_incarnation_N
+# 2018-04-06T08:21:37.142697_incarnation_N.zip
+_ARCHIVE_PATTERNS_DIRECTORY = re.compile(r"^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d+_incarnation_(\d+)$")
```

Comment:
For the transitionary phase, do we need to care about the archives created by the older agents? Or can we ignore those for now? Maybe you can make the regex conditional to look for archives that have `_incarnation_(\d+)` and also at the ones that dont have it (the older ones)

Reply:
Good point, we only purge every 24 hours so there can definitely be some leftovers.
Let me make the incarnation addition optional.